db/db_impl/db_impl.cc

```diff
@@ -530,6 +530,37 @@ Status DBImpl::MaybeReleaseTimestampedSnapshotsAndCheck() {
   return Status::OK();
 }
 
+void DBImpl::UntrackDataFiles() {
+  auto sfm = static_cast<SstFileManagerImpl*>(
+      immutable_db_options_.sst_file_manager.get());
+  assert(sfm);
+  std::vector<ColumnFamilyMetaData> metadata;
+  GetAllColumnFamilyMetaData(&metadata);
+
+  for (const auto& md : metadata) {
+    for (const auto& lmd : md.levels) {
+      for (const auto& fmd : lmd.files) {
+        // We're assuming that each sst file name exists in at most one of
+        // the paths.
+        std::string file_path =
+            fmd.directory + kFilePathSeparator + fmd.relative_filename;
+        sfm->OnDeleteFile(file_path).PermitUncheckedError();
```

Comment:
Ideally, this function should be deduplicated with the `TrackinExistingDataFiles` one since they look so similar. Maybe making `TrackingExistingDataFiles` calling this one, and pass some arguments to indicate whether it needs to track or untrack?